Python/Product/VSInterpreters/CondaEnvironmentFactoryProvider.cs

```diff
@@ -0,0 +1,372 @@
+            public string[] EnvironmentRootFolders = null;
+        }
+
+        private List<PythonInterpreterInformation> FindCondaEnvironments(string condaPath) {
+            var found = new List<PythonInterpreterInformation>();
+            var watchFolders = new HashSet<string>();
+
+            // Find environments that were created with "conda create -n <name>"
+            var condaInfoResult = ExecuteCondaInfo(condaPath);
+            if (condaInfoResult != null) {
+                foreach (var folder in condaInfoResult.EnvironmentFolders) {
+                    if (!Directory.Exists(folder)) {
+                        continue;
+                    }
+
+                    PythonInterpreterInformation env = CreateEnvironmentInfo(folder);
+                    if (env != null) {
+                        found.Add(env);
+                    }
+                }
+            }
+
+            // Find environments that were created with "conda create -p <folder>"
+            // Note that this may have a bunch of entries that no longer exist
+            // as well as duplicates that were returned by conda info.
+            if (File.Exists(_environmentsTxtPath)) {
+                try {
+                    var folders = File.ReadAllLines(_environmentsTxtPath);
+                    foreach (var folder in folders) {
+                        if (!Directory.Exists(folder)) {
+                            continue;
+                        }
+
+                        if (found.FirstOrDefault(pii => PathUtils.IsSameDirectory(pii.Configuration.PrefixPath, folder)) != null) {
+                            continue;
+                        }
+
+                        PythonInterpreterInformation env = CreateEnvironmentInfo(folder);
+                        if (env != null) {
+                            found.Add(env);
+                        }
+                    }
+                } catch (IOException) {
+                } catch (UnauthorizedAccessException) {
+                }
+            }
+
+            return found;
+        }
+
+        private static PythonInterpreterInformation CreateEnvironmentInfo(string prefixPath) {
+            var name = Path.GetFileName(prefixPath);
+            var description = name;
+            var vendor = string.Empty;
+            var vendorUrl = string.Empty;
+            var supportUrl = string.Empty;
+            var interpreterPath = Path.Combine(prefixPath, CondaEnvironmentFactoryConstants.ConsoleExecutable);
+            var windowsInterpreterPath = Path.Combine(prefixPath, CondaEnvironmentFactoryConstants.WindowsExecutable);
+
+            InterpreterArchitecture arch = InterpreterArchitecture.Unknown;
+            Version version = null;
+
+            if (File.Exists(interpreterPath)) {
+                using (var output = ProcessOutput.RunHiddenAndCapture(
+                    interpreterPath, "-c", "import sys; print('%s.%s' % (sys.version_info[0], sys.version_info[1]))"
+                )) {
```

Comment:
In theory, we could query `conda` for this information, right?

Reply:
We could query the list of packages using conda.exe, and see which version of python is installed, but that would be much slower than the above code. Also, I was curious and checked the file properties on conda.exe and turns out there's no version resources at all on it (not that I would want to use that here but I was hoping to use it somewhere else)

Reply:
You're right. Ultimately less reliable too.

I'm pretty sure conda.exe is just a standard script wrapper generated by setuptools or a similar library (distlib is common), so it's not going to have any normal features. It's on my list of "projects I'd love to have time to create"